addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -321,7 +321,13 @@ def _compute_from_holiday_status_id(self):
         )
         accruals_dict = {time_off_type.id: ids for time_off_type, ids in accruals_read_group}
         for allocation in self:
-            allocation.number_of_days = allocation.number_of_days_display
+            if allocation.holiday_status_id.request_unit != 'hour':
+                allocation.number_of_days = allocation.number_of_days_display
+            else:
+                allocation_calendar = allocation.holiday_status_id.company_id.resource_calendar_id
+                if allocation.holiday_type == 'employee':
+                    allocation_calendar = allocation.employee_id.sudo().resource_calendar_id
+                allocation.number_of_days = allocation.number_of_hours_display / (allocation_calendar.hours_per_day or HOURS_PER_DAY)
```

Comment:
I don't really understand what's happening here, isn't this code partially duplicating the code right below? Can we look at this together when you have time?

Reply:
Changed. For some reason, the method sometimes catches `type_request_unit` is sometimes caught False, hence the change for `holiday_status_id.request_unit`.